### PR TITLE
Added a default to encap to ensure TF's idempotent behaviour in the resource 'l3out_floating_svi'

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

**Review by anvitha-jain — DISMISSED** [review]

**Review by gmicol — APPROVED** [review]

File: aci/resource_aci_l3extvirtuallifp.go [hunk 1]
```diff
@@ -209,7 +210,32 @@ func resourceAciVirtualLogicalInterfaceProfile() *schema.Resource {
...
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, m interface{}) error {
+			configOld, configNew := diff.GetChange("relation_l3ext_rs_dyn_path_att")
+			updatedConfig := compareConfigs(configOld.(*schema.Set).List(), configNew.(*schema.Set).List())
+			diff.SetNew("relation_l3ext_rs_dyn_path_att", updatedConfig)
+			return nil
+		},
+	}
+}
+
+func compareConfigs(old []interface{}, new []interface{}) []interface{} {
```
Comment (sajagana) [inline aci/resource_aci_l3extvirtuallifp.go:222]:
```suggestion
func compareL3ExtRsDynPathAtt(old []interface{}, new []interface{}) []interface{} {
```

File: aci/resource_aci_l3extvirtuallifp.go [hunk 2]
```diff
@@ -209,7 +210,32 @@ func resourceAciVirtualLogicalInterfaceProfile() *schema.Resource {
 				},
 			},
 		}),
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, m interface{}) error {
+			configOld, configNew := diff.GetChange("relation_l3ext_rs_dyn_path_att")
+			updatedConfig := compareConfigs(configOld.(*schema.Set).List(), configNew.(*schema.Set).List())
```
Comment (sajagana) [inline aci/resource_aci_l3extvirtuallifp.go:215]:
```suggestion
			updatedConfig := compareL3ExtRsDynPathAtt(configOld.(*schema.Set).List(), configNew.(*schema.Set).List())
```

**Review by sajagana — CHANGES_REQUESTED** [review]
Add the missing attributes to the data source.

File: aci/resource_aci_l3extvirtuallifp.go (hunk 2 above)
Comment (shrsr) [reply aci/resource_aci_l3extvirtuallifp.go]: ok

File: aci/resource_aci_l3extvirtuallifp.go (hunk 1 above)
Comment (shrsr) [reply aci/resource_aci_l3extvirtuallifp.go]: ok

File: aci/resource_aci_l3extvirtuallifp.go [hunk 3]
```diff
@@ -209,7 +210,31 @@ func resourceAciVirtualLogicalInterfaceProfile() *schema.Resource {
...
+		for _, new_map := range new {
+			if old_map.(map[string]interface{})["tdn"] == new_map.(map[string]interface{})["tdn"] {
+				for key, value_old := range old_map.(map[string]interface{}) {
+					if key == "encap" {
+						if value_old == "unknown" && new_map.(map[string]interface{})[key] == "" {
+							new_map.(map[string]interface{})["encap"] = "unknown"
+							break
+						}
+					}
+				}
```
Comment (akinross) [inline aci/resource_aci_l3extvirtuallifp.go:233]: Do we need this loop? Can't we get the encap value from old_map, and new_map directly?
```suggestion
				if old_map.(map[string]interface{})["encap"] == "unknown" && new_map.(map[string]interface{})["encap"] == "" {
					new_map.(map[string]interface{})["encap"] = "unknown"
					break
				}
```

**Review by akinross — CHANGES_REQUESTED** [review]

**Review by akinross — APPROVED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Pull request merged**